File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/ui/connection/ConnectionActivity.java
package com.thibaudperso.sonycamera.timelapse.ui.connection;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import com.thibaudperso.sonycamera.R;
import com.thibaudperso.sonycamera.timelapse.ui.SingleFragmentActivity;

public class ConnectionActivity extends SingleFragmentActivity {


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mNotConnectedMessage = false;
    }

    @Override
    protected Fragment createFragment() {
        return new ConnectionFragment();
    }

    @Override
    protected String getGuideTitle() {
        return getString(R.string.title_connection);
    }

    @Override
    public void onBackPressed() {
        mApplication.getCameraAPI().closeConnection();
        mApplication.getWifiHandler().disconnect();
        mStateMachineConnection.reset();
        finish();
    }
}

File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/ui/connection/ConnectionActivity.kt
package com.thibaudperso.sonycamera.timelapse.ui.connection

import android.os.Bundle
import com.thibaudperso.sonycamera.R
import com.thibaudperso.sonycamera.timelapse.ui.SingleFragmentActivity
import io.reactivex.schedulers.Schedulers

class ConnectionActivity : SingleFragmentActivity<ConnectionFragment>() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        notConnectedMessage = false
    }

    override fun onBackPressed() {
        if (cameraAPI.initialized) {
            cameraAPI.closeConnection().subscribeOn(Schedulers.io()).subscribe()
        }
        wifiHandler.disconnect()
        stateMachineConnection.reset()
        finish()
    }

    override fun createFragment() = ConnectionFragment()

    override val guideTitle: String
        get() = getString(R.string.title_connection)
}
